### DISCUSSION TITLE
looping timeline help!

### DISCUSSION THREAD

**Discussion by jshinnn** [jspsych/jsPsych#2821] [Q&A]
is there a way to do this with the looping timeline?

1. present instructions pages
2. present comprehension check pages
3. if not all comprehension questions are answered correctly, present a `jsPsychHtmlButtonResponse` page that says something like “_**Unfortunately, you missed some of the comprehension questions. Please review the instructions again.**_” and redirects the participant back to the instructions and comprehension check pages

- if answered correctly, go to the second part of the experiment
- if not answered correctly the first few tries, but finally the user gets it right, move on to the second part of the experiment.

currently, this is what i have, but i don’t know how to present the `jsPsychHtmlButtonResponse` page at the appropriate time (and especially when the user doesnt get it right the first or second time, but does the subsequent try)

```
const loop_node = {
    timeline: [
        instructions,
        comprehension_check,
    ],
    loop_function: function (data) {
        var trial_data = jsPsych.data.getLastTrialData().filter({ trial_type: 'survey-multi-choice' }).trials[0]["response"];
        if (JSON.stringify(instructions) === JSON.stringify(instructions_key)) {
            return false;
        } else {
            return true;
        }
    }
}
```

**Comment by jodeleeuw** [comment]
> currently, this is what i have, but i don’t know how to present the jsPsychHtmlButtonResponse page at the appropriate time (and especially when the user doesnt get it right the first or second time, but does the subsequent try)

Check out the [conditional_function](https://www.jspsych.org/7.3/overview/timeline/#conditional-timelines) option for timelines. You could nest a timeline in your `loop_node` timeline, after the `comprehension_check`, that only runs in the event of failing the comprehension check.

If you need more guidance feel free to follow up here!
